cdap-ui/app/cdap/components/NamespaceDropdown/NamespaceDropdown.scss

```diff
@@ -69,23 +69,30 @@ $star-color: #373a3c;
       padding-right: 10px;
     }
     .namespace-text {
-      width: 120px;
-      height: 50px;
-      text-overflow: ellipsis;
-      white-space: nowrap;
-      overflow: hidden;
-      display: inline-flex;
-      flex-direction: column;
-      float: left;
+      line-height: 16px;
+      height: 16px;
+      padding-top: 2px;
 
       small {
-        line-height: 0.5;
-        padding-top: 9px;
         color: gray;
         font-size: 10px;
       }
+    }
+
+    .namespace-and-caret {
+      display: flex;
+      align-items: center;
+
       span {
         line-height: 1.5;
+        overflow: hidden;
+        text-overflow: ellipsis;
+        white-space: nowrap;
+      }
```

Comment:
does this work without explicit width for the span?

Reply:
Yep, tested it out and it's working